packages/components/src/internal/components/permissions/GroupDetailsPanel.tsx

```diff
@@ -2,38 +2,122 @@
+    const [created, setCreated] = useState<string>('');
+
+    const loadWhenCreated = useCallback(async () => {
+        try {
+            const createdState = await getWhenCreated(principal.get('userId'));
+            setCreated(createdState.slice(0, -7));
+        } catch (e) {
+            console.error(resolveErrorMessage(e) ?? 'Failed to when group created');
+        }
+    }, [principal]);
 
+    useEffect(() => {
+        loadWhenCreated();
+    }, [loadWhenCreated]);
+
+    const renderUserProp = useCallback((title: string, prop: string) => {
         return (
-            <Panel>
-                <Panel.Heading>Group Details</Panel.Heading>
-                <Panel.Body>
-                    {principal ? (
-                        <>
-                            <p className="principal-title-primary">{principal.displayName}</p>
-                            <EffectiveRolesList {...this.props} userId={principal.userId} />
-                            {/* TODO when groups are implemented, "Members" for groups*/}
-                        </>
-                    ) : (
-                        <div>No group selected.</div>
-                    )}
-                </Panel.Body>
-            </Panel>
+            <Row>
+                <Col xs={4} className="group-detail-label">
+                    {title}:
+                </Col>
+                <Col xs={8} className="group-detail-value">
+                    {prop}
+                </Col>
+            </Row>
         );
-    }
-}
+    }, []);
+
+    const renderMembersList = useCallback(() => {
+        return members.length === 0 ? (
+            <></>
+        ) : (
+            <>
+                <hr className="group-hr" />
+                <div className="group-detail-label">Members:</div>
+                <ul className="groups-ul">
+                    {members.map(member => (
+                        <li key={member.id}>{member.name}</li>
+                    ))}
+                </ul>
+            </>
+        );
+    }, [members]);
+
+    const { usersCount, groupsCount } = useMemo(() => {
+        const usersCount = members.filter(member => member.type === 'u').length;
+        const groupsCount = (members.length - usersCount).toString();
+
+        return { usersCount, groupsCount };
+    }, [members]);
+
+    return (
+        <Panel>
+            <Panel.Heading>Group Details</Panel.Heading>
+            <Panel.Body>
+                {principal ? (
+                    <>
+                        <p className="principal-title-primary">{principal.displayName}</p>
+
+                        {renderUserProp('User Count', usersCount.toString())}
+                        {renderUserProp('Group Count', groupsCount)}
+
+                        <hr className="group-hr" />
+                        {renderUserProp('Created', created)}
+
+                        <EffectiveRolesList {...props} userId={principal.userId} />
+                        {renderMembersList()}
```

Comment:
`renderMembersList` should be a component, or the logic should just be embedded directly here. In general, a method called `renderX` should always be a component. Components are easier to test (since they can be tested in an isolated fasion, allowing us to reduce the complexity of the tests that use the component), and allow react to optimize DOM updates (because we can do stuff like wrap them in `memo` and reduce unnecessary renders).